docs/guide/website/versioned_docs/version-12.20.2/infrastructure/docker.md

```diff
@@ -111,7 +115,8 @@ bash -c "./duckling & ./bp"
 
```

Comment:
Above in line 107/111 DUCKLINGURL is still missing

Reply:
It turns out that the `BP_MODULE_NLU_DUCKLINGURL` defaults to `http://localhost:8000` when running a container using the `botpress/server:{tag}` Docker image (see https://github.com/botpress/botpress/blob/master/build/docker/Dockerfile#L20)

For this reason, specifying `-e BP_MODULE_NLU_DUCKLINGURL=http://localhost:8000` when running a container using the `botpress/server:{tag}` is redundant.

The documentation should not instruct the user to specify `BP_MODULE_NLU_DUCKLINGURL`. For example, this example of documentation:

```
docker run -d \
--name bp \
-p 3000:3000 -p 3100:3100 \
-v botpress_data:/botpress/data \
-e BP_MODULE_NLU_DUCKLINGURL=http://localhost:8000 \
-e BP_MODULE_NLU_LANGUAGESOURCES='[{ "endpoint": "http://localhost:3100" }]' \
botpress/server:$TAG \
bash -c "./duckling & ./bp lang --langDir /botpress/data/embeddings & ./bp"
```

Should become

```
docker run -d \
--name bp \
-p 3000:3000 -p 3100:3100 \
-v botpress_data:/botpress/data \
-e BP_MODULE_NLU_LANGUAGESOURCES='[{ "endpoint": "http://localhost:3100" }]' \
botpress/server:$TAG \
bash -c "./duckling & ./bp lang --langDir /botpress/data/embeddings & ./bp"
```

What do you think @JustusNBB?

Reply:
Hm indeed when you run them in the same container, it's redundant to overwrite the default. But with multiple containers this needs to be overwritten. IMHO there is no harm done in having it in too many places, because as soon as you split things up you need may need to reconfigure. Rather list all endpoint wirings, especially if they don't have the same format!

Reply:
@spg EXTERNAL_URL, DATABASE_URL, DATABASE_POOL and REDIS_URL might also fit into this category, at least I have those too in my docker-compose.yml

Reply:
I get your point about listing all endpoint wirings and being explicit about the ports.

However, I think that adding the `BP_MODULE_NLU_LANGUAGESOURCES` or `BP_MODULE_NLU_DUCKLINGURL` when not needed (e.g. when running a single container all 3 processes: `bash -c "./duckling & ./bp lang --langDir /botpress/data/embeddings & ./bp"`) will confuse the end user in thinking that he needs to specify both environment variables. 

I think it should be clear to the Botpress user that both environment variables need only to be used when needed (multi-container setup, or when processes use ports that are not default)